### PR TITLE
chore: move to dax

### LINKED ISSUE #39: use dax for patch.ts

### PR DESCRIPTION
**Pull request by lino-levan**
closes #39

### CODE REVIEW

**Comment by lino-levan** [comment]
This unfortunately requires us to loosen permissions because dax but this is a dev script anyways so it's not a huge deal.

**Comment by crowlKats** [comment]
hm, if this is all the improvement we get out of switching to dax (additionally, we should also do all the things we do in the deno.json script in the patch.ts file with dax instead of via the script), then i dont see much point.

**Comment by lino-levan** [comment]
Ah I misunderstood your proposal, let me update this.

**Comment by crowlKats** [comment]
Even with updating from the deno.json script, i wonder if it makes sense to do switch

**Comment by lino-levan** [comment]
@crowlKats PTAL!

File: style/patch.ts [hunk 1]
```diff
@@ -82,11 +84,11 @@ for (const selector of KATEX_CSS_SELECTORS) {
 // de-duplicate classes
 classes = [...new Set(classes)];
 
-console.log("Writing the final style.js");
-const CSS = await Deno.readTextFile("./dist/main.css");
+$.logStep("Writing the final style.js");
+const CSS = await Deno.readTextFile("./style/dist/main.css");
```
Comment (dsherret) [inline style/patch.ts:88]:
```suggestion
const CSS = await cwd.join("./style/dist/main.css").readText();
```

File: style/patch.ts [hunk 2]
```diff
@@ -82,11 +84,11 @@ for (const selector of KATEX_CSS_SELECTORS) {
...
 
 await Deno.writeTextFile(
```
Comment (dsherret) [inline style/patch.ts:90]:
```suggestion
await cwd.join("./style.js").writeText(
```

File: style/patch.ts [hunk 3]
```diff
@@ -82,11 +84,11 @@ for (const selector of KATEX_CSS_SELECTORS) {
...
-  "style.js",
+  "./style.js",
```
Comment (dsherret) [inline style/patch.ts:91]:
```suggestion
```

File: style/patch.ts [hunk 4]
```diff
@@ -1,23 +1,29 @@
...
-    { path: "main.scss" },
-    ...expandGlobSync("node_modules/@primer/css/markdown/*.scss"),
-  ]
-) {
-  const file = Deno.readTextFileSync(path);
+const cwd = $.path(".");
+const scssFiles = [
+  cwd.join("./style/main.scss"),
+  ...Array.from(
+    cwd.expandGlobSync("node_modules/@primer/css/markdown/*.scss"),
```
Comment (dsherret) [inline style/patch.ts:14]: Actually, does this match any files? Maybe it should be `cwd.expandGlobSync("styles/node_modules/@primer/css/markdown/*.scss")`?

File: style.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
...en.string,.gfm-highlight .token.string{color:var(--color-prettylights-syntax-string)}.markdown-body .highlight .token.comment,.gfm-highlight .token.comment{color:var(--color-prettylights-syntax-comment)}.markdown-body .highlight .token.class-name,.gfm-highlight .token.class-name{color:var(--color-prettylights-syntax-variable)}.markdown-body .highlight .token.regex,.gfm-highlight .token.regex{color:var(--color-prettylights-syntax-string)}.markdown-body .highlight .token.regex .regex-delimiter,.gfm-highlight .token.regex .regex-delimiter{color:var(--color-prettylights-syntax-constant)}.markdown-body .highlight .token.tag .token.tag,.markdown-body .highlight .token.property,.gfm-highlight .token.tag .token.tag,.gfm-highlight .token.property{color:var(--color-prettylights-syntax-entity-tag)}.markdown-body .highlight .token.deleted,.gfm-highlight .token.deleted{color:var(--color-prettylights-syntax-markup-deleted-text);background-color:var(--color-prettylights-syntax-markup-deleted-bg)}.markdown-body .highlight .token.inserted,.gfm-highlight .token.inserted{color:var(--color-prettylights-syntax-markup-inserted-text);background-color:var(--color-prettylights-syntax-markup-inserted-bg)}`;
```
Comment (dsherret) [inline style.js:3]: Has the output changed? Maybe this has introduced a bug? (Maybe the glob change I mention below, which I think might have been a bug in the code I suggested)

**Review by dsherret — COMMENTED** [review]
Thanks a lot for doing this! It's looking good. I added a few comments.

File: style/patch.ts (hunk 4 above)
Comment (lino-levan) [reply style/patch.ts:14]: I think you might be right... let me look at this again.

File: style.js (hunk 5 above)
Comment (lino-levan) [reply style.js:3]: I thought that was weird, probably should have investigated. Will return with my findings, probably the bug you pointed out above.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks a lot @lino-levan!

**Pull request merged**